lib/ansible/modules/remote_management/intersight/intersight_facts.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: intersight_facts
+short_description: Gather facts about Intersight
+description:
+- Gathers facts about servers in L(Cisco Intersight,https://intersight.com).
+extends_documentation_fragment: intersight
+options:
+  server_names:
+    description:
+    - Server names to retrieve facts from.  An empty list will return all servers.
+    required: yes
+author:
+- David Soper (@dsoper2)
+- CiscoUcs (@CiscoUcs)
+version_added: '2.8'
+'''
+
+EXAMPLES = r'''
+- name: get facts for all servers
+  intersight_facts:
+    api_private_key: ~/Downloads/SecretKey.txt
+    api_key_id: 64612d300d0982/64612d300d0b00/64612d300d3650
+    server_names:
```

Comment:
Server names is empty ?

Reply:
I was following how the AWS ec2_instance_facts module behaves where an empty list returns all instances.